python_modules/dagster/dagster_tests/asset_defs_tests/partition_mapping_tests/test_asset_partition_mappings.py

```diff
@@ -586,3 +592,147 @@ def test_exported_partition_mappings_whitelisted():
     } - {PartitionMapping}
 
     assert set(get_builtin_partition_mapping_types()) == exported_partition_mapping_classes
+
+
+def test_multipartitions_def_partition_mapping_infer_identity():
+    composite = MultiPartitionsDefinition(
+        {
+            "abc": StaticPartitionsDefinition(["a", "b", "c"]),
+            "123": StaticPartitionsDefinition(["1", "2", "3"]),
+        }
+    )
+
+    @asset(partitions_def=composite)
+    def upstream(context):
+        return 1
+
+    @asset(partitions_def=composite)
+    def downstream(context, upstream):
+        assert (
+            context.asset_partition_keys_for_input("upstream")
+            == context.asset_partition_keys_for_output()
+        )
+        return 1
+
+    assets_job = define_asset_job("foo", [upstream, downstream]).resolve([upstream, downstream], [])
+    asset_graph = AssetGraph.from_assets([upstream, downstream])
+
+    assert (
+        asset_graph.get_partition_mapping(upstream.key, downstream.key)
+        == IdentityPartitionMapping()
+    )
+    assert assets_job.execute_in_process(
+        partition_key=MultiPartitionKey({"abc": "a", "123": "1"})
+    ).success
+
+
+def test_multipartitions_def_partition_mapping_infer_single_dim_to_multi():
+    abc_def = StaticPartitionsDefinition(["a", "b", "c"])
+
+    composite = MultiPartitionsDefinition(
+        {
+            "abc": abc_def,
+            "123": StaticPartitionsDefinition(["1", "2", "3"]),
+        }
+    )
+
+    @asset(partitions_def=abc_def)
+    def upstream(context):
+        assert context.asset_partition_keys_for_output("result") == ["a"]
+        return 1
+
+    @asset(partitions_def=composite)
+    def downstream(context, upstream):
+        assert context.asset_partition_keys_for_input("upstream") == ["a"]
+        assert context.asset_partition_keys_for_output("result") == [
+            MultiPartitionKey({"abc": "a", "123": "1"})
+        ]
+        return 1
+
+    asset_graph = AssetGraph.from_assets([upstream, downstream])
+
+    assert (
+        asset_graph.get_partition_mapping(upstream.key, downstream.key)
+        == MultiToSingleDimensionPartitionMapping()
+    )
+
+    class MyIOManager(IOManager):
+        def handle_output(self, context, obj):  # pylint: disable=unused-argument
+            if context.asset_key == AssetKey("upstream"):
+                assert context.has_asset_partitions
+                assert context.asset_partition_keys == ["a"]
+
+        def load_input(self, context):
+            assert context.has_asset_partitions
+            assert context.asset_partition_keys == ["a"]
+
+    upstream_job = build_assets_job(
```

Comment:
Not blocking, but I think we should use `materialize` instead of `build_assets_job` in this kind of test